### PR TITLE
Add documentation for  experimental components

### LINKED ISSUE #1997: [Repo] Codify the stages for experimental components
Document some of the outcomes from the discussion in this [thread](https://github.com/carbon-design-system/carbon-addons-iot-react/issues/1137) about experimental component stages. 

There will be further edits made in this issue.

### PR DESCRIPTION
**Pull request by davidicus**
Closes #1997 

**Summary**

Added docs explaining what an 'experimental component' means in regards to the PAL. This document outlines why components are added as experimental and how to get them out.

I also remove any components from experimental section that did not need to be there. 

**Change List (commits, features, bugs, etc)**

- added experimental-components.md to our docs folder
- removed experimental from story title for all components no longer needed in experimental
- wrote some additional test for DateTimePicker.jsx
- added icons to stories for components that are deprecated, experimental, or have insufficient test.
- changed welcome story to include legend for icons.
- updated our jest.config.js file to update thresholds and remove items no longer needed

**Acceptance Test (how to verify the PR)**

- verify that stories that used to be in experimental in [next](https://next.carbon-addons-iot-react.com/?path=/story/*) are located with regular stories in [deploy preview](https://deploy-preview-2064--carbon-addons-iot-react.netlify.app/)

### CODE REVIEW

File: packages/react/src/components/DashboardEditor/DashboardEditor.jsx [hunk 1]
```diff
@@ -372,6 +373,14 @@ const DashboardEditor = ({
   // eslint-disable-next-line react/prop-types
   onFetchDynamicDemoHotspots, // needed for the HotspotEditorModal, see the proptypes for more details
 }) => {
+  React.useEffect(() => {
```
Comment (sstone2423) [inline packages/react/src/components/DashboardEditor/DashboardEditor.jsx:376]: should we swap all of the components to show their warnings like this? It does make more sense to only display it once

File: packages/react/src/utils/__tests__/__snapshots__/publicAPI.test.js.snap [hunk 2]
```diff
@@ -18197,7 +18201,9 @@ Map {
       "wrapperClassName": null,
     },
     "propTypes": Object {
-      "addToList": [Function],
```
Comment (sstone2423) [inline packages/react/src/utils/__tests__/__snapshots__/publicAPI.test.js.snap:18200]: Do you know why this one changed?

**Review by sstone2423 — APPROVED** [review]
Thanks!

File: packages/react/src/utils/__tests__/__snapshots__/publicAPI.test.js.snap (hunk 2 above)
Comment (davidicus) [reply packages/react/src/utils/__tests__/__snapshots__/publicAPI.test.js.snap:18200]: it was marked as experimental feature and was wrapped by a warning function

File: packages/react/docs/guides/experimental-components.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+## Are these components safe to use?
+
+You should only use experimental components in development. Without proper test in place to ensure functionality and resiliency these components could introduce bugs and negatively impact your user's experience.
+
+If your application is dependent on a component that has experimental status please elevate this in our PAL priority call.
+
+## The road to stable
+
+The process of moving a component over to stable begins when there are no outstanding requirements or design questions. Once this is the case then the component is subject to our [acceptance criteria](url).
```
Comment (kevinsperrine) [inline packages/react/docs/guides/experimental-components.md:23]: The link to the acceptance criteria is missing here, but maybe just waiting until it live?

File: packages/react/docs/guides/experimental-components.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+## TL;DR
+
+Experimental components are a work in progress and not subject to the normal rules of semantic versioning. Any use of these components or features should come with the implicit understanding that minor version bumps may introduce breaking changes. These components have inherently low test coverage and documentation due to shifting apis and architecture.
+
+## What is an 'experimental component'
+
+The term experimental can mean many things in the development context. For the PAL project this term is applied to components whose designs are being validated and iterated on. During this process the API and internal markup may drastically change in response to feedback. These components have inherently low test coverage and documentation, due to this volatility,
+
+Experimental components are not subject to the normal rules of [semantic versioning](url). This means minor version bumps may introduce breaking changes.
```
Comment (kevinsperrine) [inline packages/react/docs/guides/experimental-components.md:13]: semantic versioning url missing. again, maybe just waiting until it's live?

File: packages/react/src/utils/__tests__/__snapshots__/publicAPI.test.js.snap [hunk 5]
```diff
@@ -18259,7 +18265,9 @@ Map {
       "editOptionText": Object {
         "type": "string",
       },
-      "hasMultiValue": [Function],
```
Comment (kevinsperrine) [inline packages/react/src/utils/__tests__/__snapshots__/publicAPI.test.js.snap:18262]: I assume this is the same as `addToList`?

**Review by kevinsperrine — COMMENTED** [review]
LGTM. Just a couple of clarifying comments.

**Review by kevinsperrine — APPROVED** [review]

File: packages/react/docs/guides/experimental-components.md (hunk 3 above)
Comment (davidicus) [reply packages/react/docs/guides/experimental-components.md:23]: No that was missed. good catch. thanks

File: packages/react/src/utils/__tests__/__snapshots__/publicAPI.test.js.snap (hunk 5 above)
Comment (davidicus) [reply packages/react/src/utils/__tests__/__snapshots__/publicAPI.test.js.snap:18262]: yes.

**Review by kevinsperrine — APPROVED** [review]

**Pull request merged**